nexus/src/context.rs

```diff
@@ -29,6 +29,61 @@ use std::str::FromStr;
 use std::sync::Arc;
 use uuid::Uuid;
 
+/// Indicates the kind of HTTP server.
+#[derive(Clone, Copy)]
+pub enum ServerKind {
+    /// This serves the internal API.
+    Internal,
+    /// This serves the external API over the normal public network.
+    External,
+    /// This serves the external API proxied over the technician port.
+    Techport,
+}
+
+/// The API context for each distinct Dropshot server.
+///
+/// This packages up the main server context, which is shared by all API servers
+/// (e.g., internal, external, and techport). It also includes the
+/// [`ServerKind`], which makes it possible to know which server is handling any
+/// particular request.
+#[derive(Clone)]
+pub struct ApiContext {
+    /// The kind of server.
+    pub kind: ServerKind,
+    /// Shared state available to all endpoint handlers.
+    pub context: Arc<ServerContext>,
+}
+
+impl ApiContext {
+    /// Create a new context with a rack ID and logger. This creates the
+    /// underlying `Nexus` as well.
+    pub async fn new(
```

Comment:
I think I'd name this `for_internal` (without changing the arguments). Then it's consistent with the other names and obvious that it creates a `ServerKind::Internal` context.

Reply:
Yeah, I had that originally and switched to make it clear we have to start with an internal kind. But that's obvious from the signature, I'll switch it back!

Reply:
Renamed in 85d66c0ed